frontend/src/abs/frontend/typechecker/Typing.jrag

```diff
@@ -386,6 +391,11 @@ aspect Typing {
     public interface HasTypes {
       java.util.List<Type> getTypes();
     }
+
```

Comment:
In case we remove explicit type parameters at call sites, this will need a `getType()` method as well:
`AnonymousFunctionDecl.getType() = this.getPureExp().getType();`

Similar for `NamedParFnAppParam`: look up the function by name, call `getType` on the Decl, return UnknownType if not found.